src/native/eventpipe/ep-sample-profiler.c

```diff
@@ -205,13 +205,44 @@ sample_profiler_enable (void)
 	}
 }
 
+static
+void
+ep_sample_event_pipe_callback(
+    const uint8_t* source_id,
+    unsigned long is_enabled,
+    uint8_t level,
+    uint64_t match_any_keywords,
+    uint64_t match_all_keywords,
+    EventFilterDescriptor* filter_data,
+    void* callback_data)
+{
+    if (filter_data) {
+        ep_char8_t *filter_data_char = (ep_char8_t *)((uintptr_t)(filter_data->ptr));
```

Comment:
Do I read that this is implemented as filter data that is passed to the provider?  If so, I have a bit of a concern around how to de-duplicate requests when there are multiple sessions.  Right now, this is a single value.  If we are going to allow this to be specified on provider enable, we should probably have some behavior that de-conflicts requests.  I'm not sure if that means taking the most frequent rate of all session requests, or something else.  Another option is to consider having an API outside of enablement that sets this, at which point it can remain a global, and that is expressed at the API level.